### PR TITLE
Task#6629_Init API and Fetch data of a Photo

### PR DESCRIPTION
**Pull request by NguyenQuocBao19182**
https://edu-redmine.framgia.vn/issues/6629

### CODE REVIEW

**Comment by NguyenQuocBao19182** [comment]
ready

File: app/src/main/java/com/framgia/my_editor_02/data/source/remote/ImageRemoteDataSource.java [hunk 1]
```diff
@@ -38,6 +38,7 @@ public static ImageRemoteDataSource getInstance() {
     }
 
     @Override
+
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/data/source/remote/ImageRemoteDataSource.java]: remove blank

File: app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailFragment.java [hunk 2]
```diff
@@ -9,13 +9,25 @@
...
+import com.framgia.my_editor_02.data.source.remote.ImageRemoteDataSource;
 import com.framgia.my_editor_02.databinding.FragmentPhotoDetailBinding;
+import java.util.Objects;
 
 public class PhotoDetailFragment extends Fragment {
+    private static final String EXTRA_PHOTO = "EXTRA_PHOTO";
+    private static final int DEFAULT_PAGE = 1;
+    private Photo mPhoto;
+    private PhotoDetailViewModel photoDetailViewModel;
+    private FragmentPhotoDetailBinding binding;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailFragment.java]: check convention

File: app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailFragment.java [hunk 3]
```diff
@@ -9,13 +9,25 @@
...
+import com.framgia.my_editor_02.data.source.Local.config.SharedPrefsApi;
+import com.framgia.my_editor_02.data.source.remote.ImageRemoteDataSource;
 import com.framgia.my_editor_02.databinding.FragmentPhotoDetailBinding;
+import java.util.Objects;
 
 public class PhotoDetailFragment extends Fragment {
+    private static final String EXTRA_PHOTO = "EXTRA_PHOTO";
+    private static final int DEFAULT_PAGE = 1;
+    private Photo mPhoto;
+    private PhotoDetailViewModel photoDetailViewModel;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailFragment.java]: check convention

File: app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailViewModel.java [hunk 4]
```diff
@@ -14,6 +27,41 @@ protected void onStart() {
...
+
+    public void setPhoto(Photo photo) {
+        mPhotoObservableField.set(photo);
+    }
+
+    public String getImageUrlDetail() {
+        if (mPhotoObservableField.get() != null) {
+            return Objects.requireNonNull(mPhotoObservableField.get()).getUrlImage().getSmall();
+        }
+        return "null";
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailViewModel.java]:
```suggestion
        return "";
```

File: app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailViewModel.java [hunk 5]
```diff
@@ -14,6 +27,41 @@ protected void onStart() {
...
+    }
 
+    public String getLocation() {
+        if (mPhotoObservableField.get() != null) {
+            return Objects.requireNonNull(mPhotoObservableField.get()).getLocation().getTitle();
+        }
+        return "null";
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/photoDetail/PhotoDetailViewModel.java]:
```suggestion
        return "";
```

**Comment by NguyenQuocBao19182** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**